lib/context.js

```diff
@@ -61,17 +63,28 @@ module.exports = async function getContext() {
           repository: {
             name: process.env.BITBUCKET_REPO_SLUG,
           },
-          // commits: [
-          //   {
-          //     message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
-          //       env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-          //       encoding: 'utf-8',
-          //     }).trim(),
-          //     author: {
-          //       username: actor,
-          //     },
-          //   },
-          // ],
+          commits: [
+            {
```

Comment:
per GH's docs, [for a `push` Webhook event](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#push), this is the shape:

- `after`
- `base_ref`
- `before`
- **`commits`**
- `compare`
- …

> I've been referencing [GH's `@actions/toolkit` schema for `context`](https://github.com/actions/toolkit/blob/HEAD/packages/github/src/context.ts) - and this [sample GH Action webhook "push" event payload](https://gist.github.com/GuillaumeFalourd/81f41fb8b53a0bd3d63856f919e501c2).

From [GH's docs on the `commits` key](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#push) in a push payload:

> An array of `commit` objects describing the pushed commits. (Pushed commits are all commits that are included in the `compare` between the `before` commit and the `after` commit.) The array includes a maximum of 20 commits. If necessary, you can use the [Commits API](https://docs.github.com/rest/reference/repos#commits) to fetch additional commits. This limit is applied to timeline events only and isn't applied to webhook deliveries.

<img width="484" alt="image" src="https://github.com/readmeio/readme-micro/assets/203725/06ba01d7-18ee-4b89-bed8-7571d781f872">

<hr>

what about using [Bitbucket's API endpoints for **List commits for revision**](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/#api-repositories-workspace-repo-slug-commits-revision-get)?

to more accurately mach the format of `payload.commits`, these two in particular are similar to GH's compare (`..`/`…`) and Diff APIs:

- [`GET /2.0/repositories/{workspace}/{repo_slug}/diffstat/{spec}` to **Compare two commit diff stats**](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/#api-repositories-workspace-repo-slug-diffstat-spec-get)
- [`GET /2.0/repositories/{workspace}/{repo_slug}/diff/{spec}` to **Compare two commits**](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/#api-repositories-workspace-repo-slug-diff-spec-get)

(the only required [OAuth 2.0 scope](https://developer.atlassian.com/cloud/bitbucket/scopes-for-oauth-2-3LO-and-forge-apps/) is `repository`, which Micro already requires.)

[… and others](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/):
- `GET /2.0/repositories/{BITBUCKET_REPO_SLUG}/commits/{revision}`
- `GET /2.0/repositories/{BITBUCKET_REPO_SLUG}/commits/{before}?include=foo&exclude=master`
- `GET /2.0/repositories/{BITBUCKET_REPO_SLUG}/commits/{BRANCH_NAME}?pagelen=100`
- `GET /2.0/repositories/{BITBUCKET_REPO_SLUG}/refs/tags?pagelen=100`

Reply:
We can't do that because we dont have a valid API key from the bitbucket pipeline. We're very limited in what we can do because of this.